Tests/PINCacheTests.m

```diff
@@ -877,7 +877,7 @@ - (void)testAsyncDiskInitialization
     
     testCache = [[PINDiskCache alloc] initWithName:cacheName];
     //This should not return until *after* disk cache directory has been created
-    [testCache objectForKey:@"some bogus key"];
```

Comment:
This actually shouldn't be changed, it failed because the optimization above, without the lock, isn't valid. I think it would cause problems for systems requesting objects early in their lifecycle which might be critical to pull from the cache.